packages/block-library/src/search/edit.js

```diff
@@ -408,15 +410,34 @@ export default function SearchEdit( {
 			</BlockControls>
 
 			<InspectorControls>
-				<PanelBody title={ __( 'Settings' ) }>
-					<VStack
-						className="wp-block-search__inspector-controls"
-						spacing={ 4 }
+				<ToolsPanel
+					label={ __( 'Settings' ) }
+					resetAll={ () => {
+						setAttributes( {
+							width: undefined,
+							widthUnit: undefined,
+						} );
+					} }
+					dropdownMenuProps={ dropdownMenuProps }
+				>
+					<ToolsPanelItem
+						hasValue={ () =>
+							PERCENTAGE_WIDTHS.includes( width ) &&
+							widthUnit === '%'
+						}
```

Comment:
```suggestion
						hasValue={ () => !! width }
```

Whatever value is set, it needs to be detected as "having a value."